lib/logstash/inputs/example.rb

```diff
@@ -28,11 +28,17 @@ def register
   end # def register
 
   def run(queue)
+    @thread = Thread.current
+
     Stud.interval(@interval) do
       event = LogStash::Event.new("message" => @message, "host" => @host)
       decorate(event)
       queue << event
+      break if stop?
```

Comment:
@purbon is right here, there is no need to break out of the interval, `Stud.stop!` will break it on the next loop.